pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java

```diff
@@ -174,37 +174,73 @@ private static void getTopicStats(Topic topic, TopicStats stats, boolean include
-                subsStats.msgRateRedeliver += cStats.msgRateRedeliver;
-                subsStats.msgRateOut += cStats.msgRateOut;
-                subsStats.msgThroughputOut += cStats.msgThroughputOut;
-                subsStats.bytesOutCounter += cStats.bytesOutCounter;
-                subsStats.msgOutCounter += cStats.msgOutCounter;
-                if (!subsStats.blockedSubscriptionOnUnackedMsgs && cStats.blockedConsumerOnUnackedMsgs) {
-                    subsStats.blockedSubscriptionOnUnackedMsgs = true;
-                }
+        if (topic instanceof PersistentTopic) {
+            tStatus.subscriptions.forEach((subName, subscriptionStats) -> {
+                stats.subscriptionsCount++;
+                stats.msgBacklog += subscriptionStats.msgBacklog;
+
+                AggregatedSubscriptionStats subsStats = stats.subscriptionStats
+                        .computeIfAbsent(subName, k -> new AggregatedSubscriptionStats());
+                subsStats.msgBacklog = subscriptionStats.msgBacklog;
+                subsStats.msgDelayed = subscriptionStats.msgDelayed;
+                subsStats.msgRateExpired = subscriptionStats.msgRateExpired;
+                subsStats.totalMsgExpired = subscriptionStats.totalMsgExpired;
+                subsStats.msgBacklogNoDelayed = subsStats.msgBacklog - subsStats.msgDelayed;
+                subsStats.lastExpireTimestamp = subscriptionStats.lastExpireTimestamp;
+                subsStats.lastAckedTimestamp = subscriptionStats.lastAckedTimestamp;
+                subsStats.lastConsumedFlowTimestamp = subscriptionStats.lastConsumedFlowTimestamp;
+                subsStats.lastConsumedTimestamp = subscriptionStats.lastConsumedTimestamp;
+                subsStats.lastMarkDeleteAdvancedTimestamp = subscriptionStats.lastMarkDeleteAdvancedTimestamp;
+                subscriptionStats.consumers.forEach(cStats -> {
+                    stats.consumersCount++;
+                    subsStats.unackedMessages += cStats.unackedMessages;
+                    subsStats.msgRateRedeliver += cStats.msgRateRedeliver;
+                    subsStats.msgRateOut += cStats.msgRateOut;
+                    subsStats.msgThroughputOut += cStats.msgThroughputOut;
+                    subsStats.bytesOutCounter += cStats.bytesOutCounter;
+                    subsStats.msgOutCounter += cStats.msgOutCounter;
+                    if (!subsStats.blockedSubscriptionOnUnackedMsgs && cStats.blockedConsumerOnUnackedMsgs) {
+                        subsStats.blockedSubscriptionOnUnackedMsgs = true;
+                    }
+                });
+                stats.rateOut += subsStats.msgRateOut;
+                stats.throughputOut += subsStats.msgThroughputOut;
             });
-            stats.rateOut += subsStats.msgRateOut;
-            stats.throughputOut += subsStats.msgThroughputOut;
-        });
+        } else {
+            ((NonPersistentTopicStatsImpl) tStatus).getNonPersistentSubscriptions()
```

Comment:
Seems a lot of duplicated codes, what's difference between them?

Reply:
Persistent Topic use `SubscriptionStatsImpl` 
 non-persistent use `NonPersistentSubscriptionStats`。
If We change the `NonPersistentSubscriptionStats`, it will invoke many change here.

Reply:
@Jason918  PTAL